### PR TITLE
feat(thumbnail)!: use new vars & create new class names

### PR DESCRIPTION
**Pull request by pfulton**
## How and where has this been tested?
 - **How this was tested:** 
 - **Browser(s) and OS(s) this was tested with:** 

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [ ] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).

- [x] This pull request is ready to merge.

### CODE REVIEW

**Comment by pfulton** [comment]
The checkerboard background is slightly darker in Express, due to the differences in the color vars used for the themes. Is this a problem? Is the checkerboard actually used in production-facing things, or is it simply for demonstration purposes?

I think the solution would be to hard-code a consistent color between the two themes, but I need some guidance/answers on those questions above.

spectrum-thumbnail
express-thumbnail

**Comment by GarthDB** [comment]
I don't think it's necessary to fix the checkerboard to be consistent for this pr, but you can double check with design in qotd if you'd like.

File: components/thumbnail/skin.css [hunk 1]
```diff
@@ -10,42 +10,51 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
   /* Add a half-percent to fix diagonal line issue in Chrome on non-retina displays */
   background-image:
-    linear-gradient(-45deg, transparent 75.5%, var(--spectrum-global-color-static-gray-500) 75.5%),
-    linear-gradient(45deg, transparent 75.5%, var(--spectrum-global-color-static-gray-500) 75.5%),
-    linear-gradient(-45deg, var(--spectrum-global-color-static-gray-500) 25.5%, transparent 25.5%),
-    linear-gradient(45deg, var(--spectrum-global-color-static-gray-500) 25.5%, transparent 25.5%);
+    linear-gradient(-45deg, transparent 75.5%, var(--spectrum-global-color-static-gray-200) 75.5%),
+    linear-gradient(45deg, transparent 75.5%, var(--spectrum-global-color-static-gray-200) 75.5%),
+    linear-gradient(-45deg, var(--spectrum-global-color-static-gray-200) 25.5%, transparent 25.5%),
+    linear-gradient(45deg, var(--spectrum-global-color-static-gray-200) 25.5%, transparent 25.5%);
```
Comment (lazd) [inline components/thumbnail/skin.css:21]: You can totally drive these with:

```
--spectrum-thumbnail-m-layer-darksquare-background-color
--spectrum-thumbnail-m-layer-lightsquare-background-color
```

**Review by lazd — CHANGES_REQUESTED** [review]
Drive checkerboard colors with the tokens and I think you've got it!

File: components/thumbnail/metadata/thumbnail.yml [hunk 2]
```diff
@@ -5,84 +5,84 @@ examples:
     description: Rectangular images will fill the entire space.
```
Comment (lazd) [inline components/thumbnail/metadata/thumbnail.yml:5]: Add a migration guide like

```
sections:
  - name: Migration Guide
    description: |
        ### T-shirt sizing
        Thumbnail now supports t-shirt sizing with the standard naming convention and requires that you specify the size by adding a `.spectrum-Thumbnail--size*` class.

      | Deprecated Classname | New Classname           |
      | -------------------- | ----------------------- |
      | `~.spectrum-Thumbnail-XXL~` | `.spectrum-Thumbnail--sizeXXL`|
      | `~.spectrum-Thumbnail-XL~` | `.spectrum-Thumbnail--sizeXL`|
      | `~.spectrum-Thumbnail-L~` | `.spectrum-Thumbnail--sizeL` |
      | `~.spectrum-Thumbnail-M~` | `.spectrum-Thumbnail--sizeM` |
      | `~.spectrum-Thumbnail-S~` | `.spectrum-Thumbnail--sizeS` |
```

**Pull request merged**